tests/rptest/tests/datalake/custom_partitioning_test.py

```diff
@@ -172,6 +174,31 @@ def describe_partitioning(self, dl, topic_name):
             itertools.dropwhile(lambda r: not r[0].startswith('# Partition'),
                                 spark_describe_out))
 
+    def partition_paths_from_files(self, spark, table_name):
+        partition_path_pattern = re.compile(
+            r".*/data/(?P<partition_path>.*)/.*parquet")
+        empty_partition_path_pattern = re.compile(r".*/data/.*parquet")
+        all_files = spark.run_query_fetch_all(
+            f"select file_path from {table_name}.files")
```

Comment:
There is also a "partition" column in the `<table>.files` table, but the inconvenient thing is that it disappears if there is no partitioning. OTOH it is nice to test the partition path functionality here.